components/tracker/src/tls.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::{
+    cell::Cell,
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use pin_project::pin_project;
+
+use crate::{slab::TrackerToken, Tracker, GLOBAL_TRACKERS, INVALID_TRACKER_TOKEN};
+
+thread_local! {
+    static TLS_TRACKER_TOKEN: Cell<TrackerToken> = Cell::new(INVALID_TRACKER_TOKEN);
+}
+
+pub fn set_tls_tracker(token: TrackerToken) {
+    TLS_TRACKER_TOKEN.with(|c| {
+        c.set(token);
+    })
+}
+
+pub fn clear_tls_tracker() {
```

Comment:
Would there be a situation the tracker is not set appropriately but `with_tls_tracker` is called later so there's no valid tracker information, or if the previous tracker information is not cleared properly? How could we find these unpaired?

Reply:
I'm not sure... Integration tests about it would be great but I'm still wondering the proper way to write them.

Reply:
It's much more difficult... There is little information along the way to check whether we are writing to the right tracker.

Reply:
Yes, it's not easy to verify the collected information itself directly, maybe we could leave this to the latter tests.